android/java/org/chromium/chrome/browser/crypto_wallet/util/Validations.java

```diff
@@ -46,100 +47,79 @@ public void validate(NetworkInfo selectedNetwork, KeyringService keyringService,
                 return;
             }
 
-            braveWalletService.getSelectedCoin(coinType -> {
-                if (coinType == CoinType.SOL) {
-                    braveWalletService.isBase58EncodedSolanaPubkey(
-                            senderAccountAddress, success -> {
-                                String responseMsg = "";
-                                if (!success) {
-                                    responseMsg = resources.getString(R.string.invalid_sol_address);
-                                }
-                                callback.call(responseMsg, success);
-                            });
-                } else if (coinType == CoinType.FIL) {
-                    if (receiverAccountAddress.isEmpty()) {
-                        return;
+            @CoinType.EnumType
+            int coinType = fromAccount.accountId.coin;
+            if (coinType == CoinType.SOL) {
+                // TODO(apaymyshev):  why are we validating wallet's account?
```

Comment:
Validation is done for showing an error message in the UI. A similar approach has been adopted for Desktop.
Here is a screenshot taken from Android:

![Screenshot_1687770950](https://github.com/brave/brave-core/assets/3308503/16c76932-98f1-4359-aa89-b66cb2b31943)

Reply:
But this validates **sender** address

Reply:
I see what you mean now!
Okay then, it's not entirely clear to me too why we need to validate the sender address.